File: app/src/main/java/br/com/popularmovies/moviedetail/viewmodel/factories/MovieDetailFactory.java
package br.com.popularmovies.moviedetail.viewmodel.factories;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import br.com.popularmovies.moviedetail.viewmodel.MovieDetailViewModel;
import br.com.popularmovies.services.movieService.MovieRepositoryImpl;

public class MovieDetailFactory implements ViewModelProvider.Factory {
    private final int movieId;
    private final MovieRepositoryImpl mMovieRepository;

    public MovieDetailFactory(MovieRepositoryImpl mMovieRepository, int movieId) {
        this.movieId = movieId;
        this.mMovieRepository = mMovieRepository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new MovieDetailViewModel(mMovieRepository, movieId);
    }
}

File: app/src/main/java/br/com/popularmovies/moviedetail/viewmodel/factories/MovieDetailFactory.kt
package br.com.popularmovies.moviedetail.viewmodel.factories

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import br.com.popularmovies.moviedetail.viewmodel.MovieDetailViewModel
import br.com.popularmovies.repositories.movie.MovieRepository
import javax.inject.Inject

class MovieDetailFactory @Inject constructor(
        private val mMovieRepository: MovieRepository,
        private val movieId: Int
) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return MovieDetailViewModel(mMovieRepository, movieId) as T
    }
}